app/lib/mavis_cli/vaccination_records/search.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+module MavisCLI
+  module VaccinationRecords
+    class Search < Dry::CLI::Command
+      desc "Search vaccination records in NHS Immunisations API for a patient"
+
+      argument :patient_id, required: true, desc: "ID of the patient"
+      argument :programme_types,
+               type: :array,
+               required: true,
+               desc: "The programme types to search for"
+      argument :date_from, required: false, desc: "Start date (YYYY-MM-DD)"
+      argument :date_to, required: false, desc: "End date (YYYY-MM-DD)"
+      argument :output_file,
+               required: false,
+               desc: "File path to save JSON output"
+
+      def call(
+        patient_id:,
+        programme_types:,
+        date_from: nil,
+        date_to: nil,
+        output_file: nil,
+        **
+      )
+        MavisCLI.load_rails
+
+        unless Flipper.enabled?(:immunisations_fhir_api_integration) &&
+                 Flipper.enabled?(:immunisations_fhir_api_integration_search)
+          puts "Cannot search: one of the feature flags is disabled " \
+                 "(immunisations_fhir_api_integration, immunisations_fhir_api_integration_search)"
+          return
+        end
+
+        patient = Patient.find_by(id: patient_id)
+        if patient.nil?
+          puts "Error: Patient with ID #{patient_id} not found"
+          return
+        end
+
+        programmes =
+          programme_types.map { |type| Programme.find_by(type: type.downcase) }
+        if programmes.any?(&:nil?)
+          puts "Error: One or more programmes not found in database; " \
+                 "available types are: #{Programme.types.keys.join(", ")}"
+          return
+        end
+
+        from_date = parse_date_opt(date_from, "DATE_FROM")
+        return if from_date == :error
+        to_date = parse_date_opt(date_to, "DATE_TO")
+        return if to_date == :error
+
+        if from_date && to_date && from_date > to_date
+          puts "Error: DATE_FROM cannot be after DATE_TO"
+          return
+        end
+
+        begin
+          bundle =
+            NHS::ImmunisationsAPI.search_immunisations(
+              patient,
+              programmes: programmes,
+              date_from: from_date,
+              date_to: to_date
+            )
+
+          if bundle.nil?
+            puts "No result returned"
+            return
+          end
+
+          json = bundle.to_json
+          pretty = JSON.pretty_generate(JSON.parse(json))
+
+          puts pretty
```

Comment:
I think if we provide an output file to write the JSON to, it shouldn't print it out. That's how the output option works normally, it's an either-or.